core/src/main/java/hudson/model/AutoCompletionCandidates.java

```diff
@@ -118,16 +119,30 @@ public void generateResponse(StaplerRequest req, StaplerResponse rsp, Object o)
             @Override
             public void onItem(Item i) {
                 String n = contextualNameOf(i);
-                if ((n.startsWith(value) || value.startsWith(n))
+                
+                //Check user's setting on whether to do case sensitive comparison, configured in user -> configure
+                //This is the same setting that is used by the global search field, should be consistent throughout
+                //the whole application.
+                boolean caseInsensitive = UserSearchProperty.isCaseInsensitive(); 
+                
+                String hay = n;
+                String needle = value;
+                
+                if(caseInsensitive) {
+                    hay = hay.toLowerCase();
+                    needle = needle.toLowerCase();
```

Comment:
Would rather this use `StringUtils#startsWithIgnoreCase` so we don't need to worry about locale.